src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -358,9 +360,20 @@ void WiFiManager::ScanDoneHandler(Platform::UniquePtr<uint8_t> data)
             return;
         }
 
-        // Internal scan is supposed to be followed by connection request
+        // Internal scan is supposed to be followed by a connection request if the SSID has been found
         if (Instance().mInternalScan)
         {
+            if (!Instance().mSsidFound && !Instance().mRecoveryTimerAborted)
+            {
+                ChipLogProgress(DeviceLayer, "No requested SSID found");
+                delete[] rawData;
```

Comment:
Hmm. Does it mean we have a leak in the line 389? How about packing `rawData` again in `UniquePtr` to prevent from leaking the data when adding new code here?